core/src/main/java/org/axonframework/queryhandling/backpressure/Backpressure.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2010-2018. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.axonframework.queryhandling.backpressure;
+
+import org.axonframework.queryhandling.UpdateHandler;
+
+/**
+ * Contract for backpressure mechanisms. Extends {@link UpdateHandler}, so it can be passed in for subscription queries.
+ * The idea is to wrap update handler and, based on specific mechanism, decide when to invoke its update method.
+ *
+ * @param <I> type of initial result
+ * @param <U> type of incremental updates
+ * @author Milan Savic
+ * @since 3.3
+ */
+public interface Backpressure<I, U> extends UpdateHandler<I, U> {
```

Comment:
Isn't `Backpressure` to generic a name for this? As you're extending from the `UpdateHandler`, this `Backpressure` contract will only work in combination with that stuff as well.
Not sure about a better name though...

Reply:
`BackpressuredUpdateHandler`?